module/VuFindSearch/src/VuFindSearch/Backend/Blender/Backend.php

```diff
@@ -0,0 +1,548 @@
+use Laminas\EventManager\EventManagerInterface;
+use VuFindSearch\Backend\AbstractBackend;
+use VuFindSearch\Backend\BackendInterface;
+use VuFindSearch\Backend\Blender\Response\Json\RecordCollection;
+use VuFindSearch\Command\SearchCommand;
+use VuFindSearch\Feature\RetrieveBatchInterface;
+use VuFindSearch\ParamBag;
+use VuFindSearch\Query\AbstractQuery;
+use VuFindSearch\Response\RecordCollectionInterface;
+
+/**
+ * Blender backend.
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class Backend extends AbstractBackend implements RetrieveBatchInterface
+{
+    /**
+     * Primary backend
+     *
+     * @var AbstractBackend
+     */
+    protected $primaryBackend;
+
+    /**
+     * Secondary backend
+     *
+     * @var AbstractBackend
+     */
+    protected $secondaryBackend;
+
+    /**
+     * Limit for number of records to blend
+     *
+     * @var int
+     */
+    protected $blendLimit;
+
+    /**
+     * Block size for interleaved records
+     *
+     * @var int
+     */
+    protected $blockSize;
+
+    /**
+     * Configuration
+     *
+     * @var \Laminas\Config\Config
+     */
+    protected $config;
+
+    /**
+     * Mappings configuration
+     *
+     * @var array
+     */
+    protected $mappings;
+
+    /**
+     * Event manager.
+     *
+     * @var EventManager
+     */
+    protected $events;
+
+    /**
+     * Constructor.
+     *
+     * @param AbstractBackend        $primary   Primary backend
+     * @param AbstractBackend        $secondary Secondary backend
+     * @param \Laminas\Config\Config $config    Blender configuration
+     * @param array                  $mappings  Mappings configuration
+     * @param EventManager           $events    Event manager
+     *
+     * @return void
+     */
+    public function __construct(
+        AbstractBackend $primary,
+        AbstractBackend $secondary,
+        \Laminas\Config\Config $config,
+        $mappings,
+        EventManager $events
+    ) {
+        $this->primaryBackend = $primary;
+        $this->secondaryBackend = $secondary;
+        $this->config = $config;
+        $this->mappings = $mappings;
+        $this->setEventManager($events);
+
+        $boost = ($this->config['Blending']['boostPosition'] ?? 0)
+            + ($this->config['Blending']['boostCount'] ?? 0);
+        $this->blendLimit = max(20, $boost);
+        $this->blockSize = $this->config['Blending']['blockSize'] ?? 10;
+    }
+
+    /**
+     * Perform a search and return record collection.
+     *
+     * @param AbstractQuery $query  Search query
+     * @param int           $offset Search offset
+     * @param int           $limit  Search limit
+     * @param ParamBag      $params Search backend parameters
+     *
+     * @return RecordCollectionInterface
+     */
+    public function search(
```

Comment:
Another very long method I'd recommend splitting up in some fashion.

Reply:
Split done.